sel/plus.hpp

```diff
@@ -14,7 +15,7 @@ namespace sel {
 ///
 template <term... Args>
   requires (sizeof...(Args) != 0)
-struct plus
+struct plus : term_base
```

Comment:
What definition of "term" do we maintain? Would it be fair to call this an expression instead? In some contexts a "term" is "2x", "3y^2", or "6(y-2)", but not "y - 2" (top level addition/subtraction).

Reply:
Do you want to say that `plus` is an `expression`, and that an `expression` is a valid `term` - where `term` is a valid argument for an `expression`? I think that's a reasonable approach and we can make the change in this PR.

I was also thinking that we'll want a concept for leaf objects in an expression tree (i.e. currently `constant` and `variable`). What do you think?

Reply:
What I originally ment to communicate was that, as far as I currently understand, some types that are currently `term` are not considerd a "term" in some contexts ("x + 1" as a whole is an expression, but not a term). My suggestion would be to simply rename term to expr for now because it is a more general concept. But perhaps `term` is used by the standard library and that is what we want to follow, like `plus` and `multiplies`?

I do not know if we need to differentiate between terms and expressions, I suppose that depends on the simplification rules and the structure that we want to have to apply those on.

Reply:
At the same time, "1(x + 1)" is a term which simplifies to "(x + 1)" so maybe the distinction doesn't matter.

Reply:
Maybe an `operation` would be a better name
https://mathvault.ca/math-glossary/#operation